pkg/controller/pod.go

```diff
@@ -249,11 +263,11 @@ func (pnc *PodNetworksController) processNextWorkItem() bool {
 			Type:        remove,
 			PodNetNS:    netnsPath,
 		})
+		results = append(results, res...)
```

Comment:
why should we append this result if there was an error handling this request ?.. I am assuming I can't trust the result.

Reply:
This is cover by the `an attachment is removed followed by a failing removal of another attachment` unit test. 

It can happen that 2 attachements have to be deleted, the first one succeed and the second one failed. The first status has still to be removed and the second one to be kept.
This fix comes together with l427, l433 and l449.